app.rb

```diff
@@ -95,11 +95,23 @@ class InvalidChannelName < StandardError; end
     redirect full_url
   end
 
+  get /(?<channel>\/[\w]+)?\/(?<project>[\w-]+)\/metadata-list/ do
```

Comment:
So I think this might just be an unfiltered query to the existing `/<CHANNEL>/<PROJECT>/metadata` endpoint.

As an example compare these potential requests:

```
# Return the metadata for the Ubuntu 12.04 64-bit build of ChefDK 0.9.0
https://omnitruck.chef.io/current/chefdk/metadata?v=0.9.0&p=ubuntu&pv=12.04&m=x86_64

# Return the metadata for all Ubuntu 12.04 builds of ChefDK 0.9.0
https://omnitruck.chef.io/current/chefdk/metadata?v=0.9.0&p=ubuntu&pv=12.04

# Return the metadata for all Ubuntu builds of ChefDK 0.9.0
https://omnitruck.chef.io/current/chefdk/metadata?v=0.9.0&p=ubuntu

# Return the metadata for all platforms of ChefDK 0.9.0
https://omnitruck.chef.io/current/chefdk/metadata?v=0.9.0
```

We can then carry over this same filtering logic to other endpoints. In particular @jaymzh wants similar behavior on `/<CHANNEL>/<PROJECT>/versions`.